crates/zed/src/languages/scala.rs

```diff
@@ -0,0 +1,61 @@
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+pub use language::*;
+use lsp::LanguageServerBinary;
+use std::{any::Any, path::PathBuf, str};
+
+pub struct MetalsLspAdapter;
+
+#[async_trait]
+impl LspAdapter for MetalsLspAdapter {
+    fn name(&self) -> LanguageServerName {
+        LanguageServerName("metals".into())
+    }
+
+    fn short_name(&self) -> &'static str {
+        "metals"
+    }
+
+    async fn fetch_latest_server_version(
+        &self,
+        _: &dyn LspAdapterDelegate,
+    ) -> Result<Box<dyn 'static + Any + Send>> {
+        Ok(Box::new(()))
+    }
+
+    async fn fetch_server_binary(
```

Comment:
Perhaps we could do what all the other editor integrations are doing (Neovim, Helix, etc.) and require the [Coursier](https://get-coursier.io/) to be installed first by the user (probably would be good to describe it in the language documentation). Then it could be used to download any desired Metals version (for example to some Zed cache directory, to not conflict with anything else). [Here's a reference from nvim-metals](https://github.com/scalameta/nvim-metals/blob/main/lua/metals/install.lua#L59), but it basically boils down to somethng like this:
```
cs bootstrap --java-opt -Xss4m --java-opt -Xms100m org.scalameta:metals_2.13:1.2.1+9-e65409f2-SNAPSHOT -r sonatype:snapshots -o ~/path/metals -f
```
:slightly_smiling_face:

Reply:
you can copy what I did on my WIP branch https://github.com/bishabosha/zed/blob/feature-add-scala/crates/zed/src/languages/scala.rs